dotnet/src/Connectors/Connectors.AzureAISearch.UnitTests/AzureAISearchVectorStoreRecordCollectionTests.cs

```diff
@@ -519,6 +526,32 @@ public async Task CanUpsertRecordWithCustomMapperAsync()
                 Times.Once);
     }
 
+    /// <summary>
+    /// Tests that the collection can be created even if the definition and the type do not match.
+    /// In this case, the expectation is that a custom mapper will be provided to map between the
```

Comment:
What will happen if custom mapper is not provided? Should we throw an exception in this case or prioritize either definition or type? We should probably add a unit test related to this scenario.

Reply:
In order to throw an exception I'd have to compare the definition with the model.  I'm trying to avoid that in order to allow the AOT scenario in future, where reflection would not be possible.  That said, some mappers will still do validation, if the mapper uses reflection to do the mapping.  So construction may still fail with a meaningful error message.  It's for cases where we don't do the mapping, or where a custom mapper is provided where those mappers should undertake this responsibility.